src/MongoDB/Manager.c

```diff
@@ -916,6 +924,41 @@ static HashTable* php_phongo_manager_get_debug_info(zval* object, int* is_temp T
 	mongoc_server_descriptions_destroy_all(sds, n);
 
 	return Z_ARRVAL(retval);
+
+failure:
+#if PHP_VERSION_ID >= 70000
+	{
+		zend_string* string_key = NULL;
+		zend_ulong   num_key    = 0;
+		zval*        server;
+
+		ZEND_HASH_FOREACH_KEY_VAL_IND(Z_ARRVAL(cluster), num_key, string_key, server)
+		{
+			zval_ptr_dtor(server);
```

Comment:
Could we get away with just calling `zval_ptr_dtor(&retval)` instead? I realize that the `ZVAL_STATIC_INIT` initializer sets everything to zero (and wouldn't identify as refcounted on its own), but [`array_init_size()`](http://lxr.jpauli.tech/source/xref/php-7.3/Zend/zend_API.h#378) will set the refcounted flag (follow `ZVAL_ARR` through to `IS_ARRAY_EX`](http://lxr.jpauli.tech/source/xref/php-7.3/Zend/zend_types.h#IS_ARRAY_EX)).

If you follow the chain from [`i_zval_ptr_dtor()`](http://lxr.jpauli.tech/source/xref/php-7.3/Zend/zend_variables.h#39) through `rc_dtor_func` (assuming we decrement the refcount to zero), you'll end up in [`zend_array_destroy()`](http://lxr.jpauli.tech/source/xref/php-7.3/Zend/zend_hash.c#1464), which will recursively call `i_zval_ptr_dtor` on each element within the array's hash table.

Reply:
Ah, good call. I tried to follow through but couldn't quite see where we end up when destroying a HashTable, so I decided to ensure we're clearing everything. I've removed the loops in favour of only calling `zval_ptr_dtor`.